w3-account.md

```diff
@@ -0,0 +1,224 @@
+# User Accounts
+
+![status:wip](https://img.shields.io/badge/status-wip-orange.svg?style=flat-square)
+[![hackmd-github-sync-badge](https://hackmd.io/8NywALT8Qp-cf0MSugZMDw/badge)](https://hackmd.io/8NywALT8Qp-cf0MSugZMDw)
+
+## Editors
+
+- [Irakli Gozalishvili](https://github.com/Gozala), [DAG House](https://dag.house/)
+
+## Authors
+
+- [Irakli Gozalishvili](https://github.com/Gozala), [DAG House](https://dag.house/)
+
+
+# Abstract
+
+
+In web3.storage we describe concept of an account as convenience for aggregating and managing capabilities across various user spaces under same identity, simplifying recovery and authorization flows.
+
+## Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC2119](https://datatracker.ietf.org/doc/html/rfc2119).
+
+
+# Introduction
+
+## Motivation
+
+In web3.storage users MAY create number of user spaces simply by generating asymmetric key pair. They MAY be delegated capabilities for other user spaces. Managing these delegations and keypairs across multiple spaces, agents and devices can get complicated.
+
+To address this we propose a concept of an account, which is a [principal][] that can be delegated all relevant capabilities across various user spaces.
+
+We also propose account authorization flow that would allowing allow authorized agent to act on behalf of the account [principal][].
+
+
+# Terminology
+
+## Account
+
+User account is a [principal][] identified by [`did:mailto`][] (listed in a UCAN's `iss` or `aud` field).
+
+When user agent creates a new space, it MAY delegate full or subset of the capabilities to user account so they could be reclaimed by a user with another agent.
+
+> Example shows all capabilities to `did:key:zAlice` space been delegated to the user account `alice@web.mail` 
+
+```ts!
+{
+  iss: "did:key:zAlice",
+  aud: "did:mailto:alice@web.mail",
+  att: [{ with: "did:key:zAlice", can: "*" }],
+  exp: null,
+  sig: "..."
+}
+```
+
+## Delegations
+
+Agent MAY publish account delegations to a serivce so that they can be persisted and retrieved later with a different agent.
+
+> Invokes `access/delegate` asking web3.storage to record delegation from `did:key:zAlice` space to the `alice@web.mail` account.
+
+ ```ts
+{
+  iss: "did:key:zAgent",
+  aud: "did:dns:web3.storage",
```

Comment:
I seem to recall somewhere you were deciding between whether to use `did:dns` or `did:web`. Where did that land? One tradeoff that comes to mind is that updates to `did:dns` may be more eventually consistent (because ttls), whereas `did:web` might give us more predictability about how fast we can expect updates to propagate.

Reply:
General idea is agnostic of choice between `did:dns` and `did:web` we could choose one that makes most sense or both. That said I think did:web is probably a going to be a better choice however